src/bin/bench-streamer.rs

```diff
@@ -54,28 +62,83 @@ fn sink(
     })
 }
 
+macro_rules! socketaddr {
+    ($ip:expr, $port:expr) => {
+        SocketAddr::from((Ipv4Addr::from($ip), $port))
+    };
+    ($str:expr) => {{
+        let a: SocketAddr = $str.parse().unwrap();
+        a
+    }};
+}
+
 fn main() -> Result<()> {
-    let read = UdpSocket::bind("127.0.0.1:0")?;
-    read.set_read_timeout(Some(Duration::new(1, 0)))?;
+    let mut num_sockets = 1usize;
+
+    let matches = App::new("solana-bench-streamer")
+        .arg(
+            Arg::with_name("num-recv-sockets")
+                .short("N")
+                .long("num-recv-sockets")
+                .value_name("NUM")
+                .takes_value(true)
+                .help("Use NUM receive sockets"),
+        )
+        .get_matches();
+
+    if let Some(n) = matches.value_of("num-recv-sockets") {
+        num_sockets = n.to_string().parse().expect("integer");
+    }
+
+    fn bind_to(port: u16) -> UdpSocket {
```

Comment:
Use the one in crdt.rs?

Reply:
That function is internal to another function (and not public). I can change that. Might be a good idea to remove duplicate code..